src/error.rs

```diff
@@ -6,73 +6,108 @@ use std::error::Error as StdError;
 #[cfg(feature = "std")]
 use self::super::TriviallyTransmutable;
 use core::fmt;
+#[cfg(feature = "std")]
+use core::marker::PhantomData;
 
 
 /// A transmutation error. This type describes possible errors originating
-/// from operations in this crate.
+/// from operations in this crate. The two type parameters, when defined,
+/// represent the source element type and the target element type
+/// respectively.
 ///
 /// # Examples
 ///
 /// ```
 /// # use safe_transmute::{ErrorReason, Error, transmute_bool_pedantic};
-/// # unsafe {
 /// assert_eq!(transmute_bool_pedantic(&[0x05]), Err(Error::InvalidValue));
-/// # }
 /// ```
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub enum Error {
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub enum Error<T = (), U = ()> {
     /// The data does not respect the target type's boundaries.
     Guard(GuardError),
     /// The given data slice is not properly aligned for the target type.
     Unaligned(UnalignedError),
-    /// The given data vector is not properly aligned for the target type.
+    /// The given data vector is not properly aligned in memory for the target type.
```

Comment:
Why "in memory"?

Reply:
Hmm, good catch. I was trying to better convey that `UnalignedVecError` was more than just a check for `align_of`s, but also that the address of the data should be in a compatible position. But regardless, we have that this is no longer returned anywhere in the code, so I might as well get rid of it. :+1: